STACK-UP-3/warriorz-backend
# #171784093 add view trip requests feature

#### What this PR does?
 -Delivers view trips' resources to the project.

#### Description of the task to be completed?

- add view all trips in a paginated list view, functionality.
- add view all trips assigned to the manager in a paginated list view, functionality.
- add view specific trip functionality.
- add validations and tests
- add swagger documentation.

#### How should this be tested?
 - First, you clone the repo on your machine.
 - Open the folder in your text editor,  *preferably   `visual studio code`*
 - Now open your terminal, *for `visual studio code`    just click on    `terminal`* 
 - Now run `npm install`
 - Then  run `npm run start:dev`,
- First `signup`, verify email, and `sign in`.
- After now create a trip request.
 - Now open `postman` and run endpoint using GET request:
> - **localhost:5000/api/v1/trips?page=`integer`&limit=`integer`** route: For viewing all trips of the user.
> - **localhost:5000/api/v1/trips/assigned?page=`integer`&limit=`integer`** route: For viewing all trips assigned to the manager.
> - **localhost:5000/api/v1/trips/`trip_id`** route: For viewing a specific trip with the specified id.

OR
- You can test it on the swagger Documentation(*in your browser*) using the `http://localhost:5000/api-docs/` route.

#### What are the relevant pivotal tracker stories?
  [#171784093](https://www.pivotaltracker.com/story/show/171784093)

#### Screenshots?
1. User View trips
![image](https://user-images.githubusercontent.com/52471122/80424697-87c45980-88e2-11ea-9988-b31a065b15d9.png)


2. Manager view assigned trips
![image](https://user-images.githubusercontent.com/52471122/80424769-a75b8200-88e2-11ea-8ee7-ef88d476fbc1.png)


3. User view a specific trip.
![image](https://user-images.githubusercontent.com/52471122/80424850-d540c680-88e2-11ea-8bc8-ec50f1ac423e.png)
